### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by Therzok**
See each commit, changes should be non-invasive.

### CODE REVIEW

File: src/ObjCRuntime/Runtime.cs [hunk 1]
```diff
@@ -1356,4 +1356,48 @@ public int GetHashCode (Type obj)
 			return obj.GetHashCode ();
 		}
 	}
+
+	internal struct IntPtrTypeValueTuple : IEquatable<IntPtrTypeValueTuple>
+	{
+		private static readonly IEqualityComparer<IntPtr> item1Comparer = Runtime.IntPtrEqualityComparer;
```
Comment (rolfbjarne) [inline src/ObjCRuntime/Runtime.cs]: Style: we don't use `private` when it's the default.

File: src/ObjCRuntime/Runtime.cs [hunk 2]
```diff
@@ -1356,4 +1356,48 @@ public int GetHashCode (Type obj)
...
+		private static readonly IEqualityComparer<Type> item2Comparer = Runtime.TypeEqualityComparer;
+
+		public readonly IntPtr Item1;
+		public readonly Type Item2;
+
+		public IntPtrTypeValueTuple(IntPtr item1, Type item2)
```
Comment (rolfbjarne) [inline src/ObjCRuntime/Runtime.cs]: Style: there should be a space before the opening parenthesis (this occurs several times in this class).

File: src/ObjCRuntime/Runtime.cs (hunk 1 above)
Comment (Therzok) [inline src/ObjCRuntime/Runtime.cs]: Fixed. For some reason force-pushing breaks reviews.

File: src/ObjCRuntime/Runtime.cs (hunk 2 above)
Comment (Therzok) [inline src/ObjCRuntime/Runtime.cs]: Fixed. For some reason force-pushing breaks reviews.

**Pull request merged**